File: dotnet/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example1/BaseOrderHandler.cs
namespace ChainOfResponsibilityDesignPattern.Example1;

public abstract class BaseOrderHandler : IOrderHandler
{
    private IOrderHandler? _nextHandler;

    public IOrderHandler SetNext(IOrderHandler handler)
    {
        _nextHandler = handler;
        return handler;
    }

    public virtual void Handle(OrderRequest request)
    {
        if (_nextHandler != null)
        {
            _nextHandler.Handle(request);
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example1/src/BaseOrderHandler.ts
import { IOrderHandler, OrderRequest } from "./OrderRequest";

export abstract class BaseOrderHandler implements IOrderHandler {
  private _nextHandler: IOrderHandler | null = null;

  setNext(handler: IOrderHandler): IOrderHandler {
    this._nextHandler = handler;
    return handler;
  }

  handle(request: OrderRequest): void {
    if (this._nextHandler) {
      this._nextHandler.handle(request);
    }
  }
}
